src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -608,6 +619,17 @@ class ParseLiveQueryServer {
     return auth;
   }
 
+  _checkFields(client: any, requestId: any, message: any) {
+    const subscriptionInfo = client.getSubscriptionInfo(requestId);
+    const fields = subscriptionInfo?.fields;
+    if (!fields) {
+      return true;
+    }
+    const object = message.currentParseObject;
+    const original = message.originalParseObject;
+    return fields.some(field => object.get(field) != original?.get(field));
```

Comment:
@dblythy does this code works correctly with a field of type Array, Object, Geo etc....

Does object and original are the same JS reference `console.log(object === original)` ?

If the reference is not the same you should use https://nodejs.org/api/util.html#utilisdeepstrictequalval1-val2

Then the equality will be correctly detected even with a JS object (Object, array, JSON etc...)

Reply:
Thanks for the suggestion :)